### PR TITLE
Add caching to RepeatController + minor fixes

### PR DESCRIPTION
**Pull request by jediandrey**
Make sure these boxes are checked before asking for review of your pull request - thank you!

If there is a new feature, please create PR against 8.x-1.xx-dev branch.

If this is a bug fix - use 8.x-1.x branch. We'll tag for release if the bug is critical asap or tag for release next bug fix release until critical issue arrived.

## Steps for review

- [ ] Please provide steps for review here.
- [ ] Please provide steps for review here.
- [ ] Please provide steps for review here.
- [ ] Please provide steps for review here.

## General checks
- [ ] All coding styles are fulfilled and there are no any issues reported by CodeSniffer CI. <br/>CI code sniffer errors
- [ ] All tests are running and there are no failed tests reported by CI. <br/>Behat test results
- [ ] [Documentation](https://github.com/ymcatwincities/openy/tree/8.x-1.x/docs) has been updated according to PR changes.
- [ ] [Steps for review](https://github.com/ymcatwincities/openy/pull/94#issue-204580200) have been provided according to PR changes. <br/>Steps for review
- [ ] Make sure you've provided all necessary hook\_update\_N to [support upgrade path](https://github.com/ymcatwincities/openy/blob/8.x-1.x/docs/Development/Upgrade%20path.md).
- [ ] Make sure your git email is associated with account on drupal.org, otherwise you won't get commits there. <br/>drupal.org email
- [ ] If you would like to get credits on drupal.org, [check documentation](https://github.com/ymcatwincities/openy/blob/8.x-1.x/docs/Development/Contributing.md#drupalorg-credits).

Thank you for your contribution!

### CODE REVIEW

**Comment by bjwilkins** [comment]
Can one of the admins verify this patch? Use "o+k to test" or ''t+est this please" for manual build execution.

File: modules/custom/openy_repeat/src/Controller/RepeatController.php [hunk 1]
```diff
@@ -167,18 +239,31 @@ public function getLocationsInfo() {
...
+            $data[$node->nid->value] = [
+              'nid' => $node->nid->value,
+              'title' => $node->title->value,
+              'description' => html_entity_decode(strip_tags(text_summary($node->field_class_description->value, $node->field_class_description->format, 600))),
+            ];
+            $tags[] = 'node:' . $node->nid->value;
+          }
+        }
+      }
+      $this->cache->set($cid, $data, CacheBackendInterface::CACHE_PERMANENT, $tags);
```
Comment (diffyDemoAccess) [inline modules/custom/openy_repeat/src/Controller/RepeatController.php]: please add tag 'node_list' so whenever new class being added this cache also will be invalidated.

File: modules/custom/openy_repeat/src/Controller/RepeatController.php [hunk 2]
```diff
@@ -136,28 +196,40 @@ public function getLocations() {
...
-
-      $data[$node->title->value] = [
-        'nid' => $node->nid->value,
-        'title' => $node->title->value,
-        'email' => $node->field_location_email->value,
-        'phone' => $node->field_location_phone->value,
-        'address' => $address,
-        'days' => !empty($days[0]) ? $this->getFormattedHours($days[0]) : [],
-      ];
+      $this->cache->set($cid, $data, CacheBackendInterface::CACHE_PERMANENT, $tags);
```
Comment (ygerasimov) [inline modules/custom/openy_repeat/src/Controller/RepeatController.php:233]: please add tag 'node_list' so whenever new class being added this cache also will be invalidated

File: modules/custom/openy_repeat/src/Controller/RepeatController.php (hunk 1 above)
Comment (ygerasimov) [inline modules/custom/openy_repeat/src/Controller/RepeatController.php:268]: please add tag 'node_list' so whenever new class being added this cache also will be invalidated

Comment (jediandrey) [reply modules/custom/openy_repeat/src/Controller/RepeatController.php:268]: done

**Review by podarok — APPROVED** [review]

**Comment by podarok** [comment]
ok to test

**Comment by bjwilkins** [comment]
Build comment file: 
--------------

 
| Environment | Link |
| ------------- | ------------- |
| Fresh OpenY installation - Rose theme | http://openy.cibox.tools/build801 |
| Fresh OpenY installation - Lily theme | http://lily.openy.cibox.tools/build801 |
| Upgraded(upgrade path) installation | http://upgrade.openy.cibox.tools/build801 |
| Installation process | http://install.openy.cibox.tools/build801/install.php |

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy.cibox.tools:8080/job/PR_BUILDER_COMPOSER/801/

**Comment by bjwilkins** [comment]
Acessibility Sniffer: front page checking WCAG2AA http://openy.cibox.tools/build801/frontWCAG2AAhtmlcs.txt
Acessibility Sniffer: join page checking WCAG2AA http://openy.cibox.tools/build801/joinWCAG2AAhtmlcs.txt
Acessibility Sniffer: locations page checking WCAG2AA http://openy.cibox.tools/build801/locationsWCAG2AAhtmlcs.txt
Acessibility Sniffer: schedules page checking WCAG2AA http://openy.cibox.tools/build801/schedulesWCAG2AAhtmlcs.txt
Acessibility Sniffer: blog page checking WCAG2AA http://openy.cibox.tools/build801/blogWCAG2AAhtmlcs.txt

**Pull request merged**